Mathlib/RingTheory/Ideal/Over.lean

```diff
@@ -37,18 +32,6 @@ section CommRing
 
 variable {S : Type*} [CommRing S] {f : R →+* S} {I J : Ideal S}
 
-/-- For a prime ideal `p` of `R`, `p` extended to `S` and
-restricted back to `R` is `p` if and only if `p` is the restriction of a prime in `S`. -/
-lemma comap_map_eq_self_iff_of_isPrime (p : Ideal R) [p.IsPrime] :
-    (p.map f).comap f = p ↔ (∃ (q : Ideal S), q.IsPrime ∧ q.comap f = p) := by
-  refine ⟨fun hp ↦ ?_, ?_⟩
-  · obtain ⟨q, hq₁, hq₂, hq₃⟩ := Ideal.exists_le_prime_disjoint _ _
-      (disjoint_map_primeCompl_iff_comap_le.mpr hp.le)
-    exact ⟨q, hq₁, le_antisymm (disjoint_map_primeCompl_iff_comap_le.mp hq₃)
-      (map_le_iff_le_comap.mp hq₂)⟩
-  · rintro ⟨q, hq, rfl⟩
-    simp
-
 theorem coeff_zero_mem_comap_of_root_mem_of_eval_mem {r : S} (hr : r ∈ I) {p : R[X]}
```

Comment:
I think this and the polynomial lemmas below could be moved into going up as well. In that case probably we won't need to import polynomial in this file?